### PR TITLE
Functionality updated for authentication

### PR DESCRIPTION
**Pull request by narendra28051991**

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -1,31 +1,43 @@
...
+  Navigate
 } from "react-router-dom";
+// pages
 import Signup from "./pages/signup/Signup";
 import Login from "./pages/login/Login";
 import Password from "./pages/password/Password";
 import Dashboard from "./pages/dashboard/Dashboard";
+// hooks
+import { useAuthContext } from "./hooks/useAuthContext";
+// styles
```
Comment (YurijPaharev) [inline frontend/src/App.tsx:17]: There is no need for these comments

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -1,31 +1,43 @@
...
+        <Route path="/signup" element={user ? <Navigate to="/dashboard" replace /> : <Signup />} />
+        <Route path="/login" element={user ? <Navigate to="/dashboard" replace /> : <Login />} />
+        <Route
+          path="/password"
+          element={user ? <Navigate to="/dashboard" replace /> : <Password />}
+        />
+        <Route path="/dashboard" element={user ? <Dashboard /> : <Navigate to="/login" replace />} />
+      </React.Fragment>
+    )
+  );
```
Comment (YurijPaharev) [inline frontend/src/App.tsx:39]: You are repeating a lot of code and with every added route you would need to do these repetitions again and again, please find another method (You can refer to the documentation https://reactrouter.com/en/main/start/tutorial#nested-routes)

File: frontend/src/hooks/useLogout.ts [hunk 3]
```diff
@@ -1,14 +1,6 @@
...
 const useLogout = () => {
-  const context = useContext(AuthContext) as AuthContextType;
-
-  if (!context) {
-    throw new Error('useLogout must be used within an AuthProvider');
-  }
-
   const logout = async () => {
     try {
       await auth.signOut();
```
Comment (YurijPaharev) [inline frontend/src/hooks/useLogout.ts:6]: This method does not return any hook methods? Probably just make it as a function?

File: frontend/src/store/login.store.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { create } from "zustand";
+import { LoginSignupState } from "../types";
+
+const useLoginState = create<LoginSignupState>((set) => ({
+  error: null,
+  setError: (error) => set({ error }),
+}));
```
Comment (YurijPaharev) [inline frontend/src/store/login.store.ts:7]: it states that it is a login store, but it stores only an error

**Review by YurijPaharev — CHANGES_REQUESTED** [review]
Very good, I especially liked documentations and .d.ts files, but please take a look at my comments

**Comment by narendra28051991** [comment]
I like these comments. I will try to solve them in a good way.

Thanks Yurii for the comments.

**Pull request closed without merging**